docs/development/testguide.rst

```diff
@@ -67,50 +69,65 @@ options require the `pytest-pep8 plugin
 <https://pypi.org/project/pytest-pep8>`_, which must be installed
 separately.
 
-setup.py test
-=============
+.. _running-pytest:
+
+pytest
+======
+
+The test suite can be run directly from the native ``pytest`` command. In this
+case, it is important for developers to be aware that they must manually
+rebuild any extensions by running::
+
+    python setup.py build_ext --inplace
 
-The astropy core package and the Astropy package template provide a ``test``
-setup command, invoked by running ``python setup.py test`` while in the
-package root directory. Run ``python setup.py test --help`` to see the
-options to the test command.
+before running the test with pytest with::
 
-Since ``python setup.py test`` wraps the widely-used pytest framework, you may
-from time to time want to pass options to the ``pytest`` command itself. For
-example, the ``-x`` option to stop after the first failure can be passed
-through with the ``--args`` argument::
+    pytest
 
-    > python setup.py test --args "-x"
+Instead of calling ``build_ext --inplace``, you can also install the package
+in editable mode::
 
-`pytest`_ will look for files that
-:ref:`look like tests <pytest:python test discovery>`, per default
-in the current directory and all recursive directories, then run all
-the code that looks like tests (essentially all functions or methods
-with ``test_`` prefixes or inside ``Test`` prefixed classes) within
-those files.
+    pip install -e .[test]
 
-Turn on PEP8 checking by passing ``--pep8`` to the ``test`` command. This will
-turn off regular testing and enable PEP8 testing.
+The latter is required if you need to test parts of the package that rely
+on certain entry points being installed.
```

Comment:
This presumes someone knows what "entry points" are - I'm afraid I don't. Is this about the C/Cython extensions?

Reply:
No, entry points are separate from extensions and are described in https://setuptools.readthedocs.io/en/latest/pkg_resources.html#entry-points - I think the best solution here is probably to just link the term 'entry points' to that setuptools docs page?

Reply:
Yes, the link would solve this.

Reply:
Done